Python/ceval.c

```diff
@@ -722,21 +724,23 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
         /* Because this avoids the RESUME,
          * we need to update instrumentation */
         _Py_Instrument(_PyFrame_GetCode(frame), tstate->interp);
-        monitor_throw(tstate, frame, frame->prev_instr);
+        monitor_throw(tstate, frame, frame->instr_ptr - 1);
         /* TO DO -- Monitor throw entry. */
         goto resume_with_error;
     }
 
     /* Local "register" variables.
      * These are cached values from the frame and code object.  */
 
-    _Py_CODEUNIT *next_instr;
+    _Py_CODEUNIT *next_instr = NULL;
```

Comment:
I assume this is because `next_instr` is set by `SET_LOCALS_FROM_FRAME()` a few lines below.
This can leave `next_instr` as `NULL` if there is a recursion error. 
I'd like to keep the invariant that `next_instr` points to the next instruction.
Can you set it before the `goto exit_unwind;` on line 748 below?